### PR TITLE
Improve Crashlytics Action Logging

### PR DESCRIPTION
**Pull request by i2amsam**
Also hides the build secret when errors occur.

Before:
image

After:
image

### CODE REVIEW

File: fastlane/spec/actions_specs/crashlytics_spec.rb [hunk 1]
```diff
@@ -41,6 +42,35 @@
...
+              build_secret: "MAJOR_KEY",
+              apk_path: "./fastlane/spec/fixtures/fastfiles/Fastfile2",
+              emails: ["email1@krausefx.com", "email2@krausefx.com"],
+              groups: "testgroup",
+              notes: "Such notes, very release"
+              )
+            end').runner.execute(:test)
+          ensure
+            $verbose = old_verbose
+          end
```
Comment (mfurtak) [inline fastlane/spec/actions_specs/crashlytics_spec.rb]: Might want to consider porting

```ruby
def with_verbose(verbose)
  orig_verbose = $verbose
  $verbose = verbose
  yield if block_given?
ensure
  $verbose = orig_verbose
end
```

from `fastlane_core`

File: fastlane/spec/actions_specs/crashlytics_spec.rb [hunk 2]
```diff
@@ -28,6 +28,7 @@
              "-apiKey api_token",
              "-apiSecret build_secret",
              "-uploadDist '/",
+             "-betaDistributionReleaseNotesFilePath '/",
```
Comment (mfurtak) [inline fastlane/spec/actions_specs/crashlytics_spec.rb]: Why did this change?

File: fastlane/lib/fastlane/actions/crashlytics.rb [hunk 3]
```diff
@@ -28,11 +28,29 @@ def self.run(params)
         end
 
         UI.success('Uploading the build to Crashlytics Beta. Time for some ☕️.')
-        UI.verbose(command.join(" ")) if $verbose
-        Actions.sh(command.join(" "), log: false)
+
+        sanitizer = proc do |message|
+          message.gsub(params[:api_token], '[[API_TOKEN]]')
+                 .gsub(params[:build_secret], '[[BUILD_SECRET]]')
```
Comment (mfurtak) [inline fastlane/lib/fastlane/actions/crashlytics.rb]: I did not know that Ruby will read onto the next line in order to determine if a statement was complete.

File: fastlane/spec/actions_specs/crashlytics_spec.rb (hunk 2 above)
Comment (i2amsam) [inline fastlane/spec/actions_specs/crashlytics_spec.rb]: Just was not being tested before

**Comment by mfurtak** [comment]
LGTM - one suggestion about porting a helper method from the `spec_helper` of `fastlane_core`

**Pull request merged**